### ISSUE TITLE
Distributor : Performance Improvement

### ISSUE THREAD

**Issue by joe-elliott** [grafana/tempo#966]
Consider vendoring Grafana's [OTEL collector fork](https://github.com/grafana/opentelemetry-collector). This will allow us access to the internal proto definitions.

Then we will be able to drop this expensive code:
https://github.com/grafana/tempo/blob/main/modules/distributor/receiver/shim.go#L169-L180

Simplify our proto build process:
https://github.com/grafana/tempo/blob/main/Makefile#L129-L168

and remove the git submodule of the otel proto.

**Comment by zalegrala** [comment]
Do we have concerns about the fork being three years old?  I don't have the context to know why we have a fork, but just wondering if perhaps there is a more active version we could vendor.

**Comment by joe-elliott** [comment]
Yeah, the original path proposed here is likely a nonstarter. It is still possible that an enterprising engineer could review the situation and find a way to avoid the marshal/unmarshal dance just to convert between the two formats. 

The link is out of date. The code is now here:
https://github.com/grafana/tempo/blob/f9000562d5606f1615af799375e89ce0ee246fd3/modules/distributor/distributor.go#L288

**Comment by joe-elliott** [comment]
Closing. We have many other opportunities for resource usage improvements before this. One day we will whittle it down and a profile will flag it as a problem.
